Main/Source/game.cpp

```diff
@@ -1324,40 +1324,70 @@ int game::RotateMapNotes()
   return iMapNotesRotation;
 }
 
-std::vector<festring> afsAutoPickupMatch;
-pcre *reAutoPickup=NULL;
-void game::UpdateAutoPickUpMatching() //simple matching syntax
+/**
+ * 
+ * @param it
+ * @param bUnarticled if false is GetNameSingular()
+ * @return 
+ */
+cchar* game::StoreMatchNameKey(item* it,bool bUnarticled)
 {
-  afsAutoPickupMatch.clear();
-
-  bool bSimple=false;
-  if(bSimple){ //TODO just drop the simple code? or start the string with something to let it be used instead of regex? tho is cool to let ppl learn regex :)
-    if(ivanconfig::GetAutoPickUpMatching().GetSize()==0 || ivanconfig::GetAutoPickUpMatching()[0]=='!')return;
+  static festring fsRet,fsSpace=" ",fsEmpty="";
+  static cchar* cToken="+";
+  
+  fsRet="";
+  fsRet<<cToken;
+  if(bUnarticled)
+    fsRet<<it->GetName(UNARTICLED|UNLABELED);
+  else
+    fsRet<<it->GetNameSingular()<<cToken;
+  
+  festring::SearchAndReplace(fsRet,fsSpace,fsEmpty,0);
+  
+  return fsRet.CStr();
+}
 
-    std::stringstream ss(ivanconfig::GetAutoPickUpMatching().CStr());
-    std::string match;
-    while(std::getline(ss,match,'|'))
-      afsAutoPickupMatch.push_back(festring(match.c_str()));
-  }else{
-    //TODO test regex about: ignoring broken lanterns and bottles, ignore sticks on fire but pickup scrolls on fire
-  //  static bool bDummyInit = [](){reAutoPickup=NULL;return true;}();
-    const char *errMsg;
-    int iErrOffset;
-    if(reAutoPickup)pcre_free(reAutoPickup);
-    reAutoPickup = pcre_compile(
-      ivanconfig::GetAutoPickUpMatching().CStr(), //pattern
-      0, //no options
-      &errMsg,    &iErrOffset,
-      0); // default char tables
-    if (!reAutoPickup){
-      std::vector<festring> afsFullProblems;
-      afsFullProblems.push_back(festring(errMsg));
-      afsFullProblems.push_back(festring()+"offset:"+iErrOffset);
-      bool bDummy = iosystem::AlertConfirmMsg("regex validation failed, if ignored will just not work at all",afsFullProblems,false);
+void game::AutoStoreItemInContainer(item* itToStore,character* C)
+{
+  if(!C->IsPlayer())
+    return;
+  
+  static itemvector vit;vit.clear();
+  C->GetStack()->FillItemVector(vit);
+  for(int i=0;i<vit.size();i++){
+    DBGITEM(vit[i],"checkingIfIsAContainer");
+    itemcontainer* itc = dynamic_cast<itemcontainer*>(vit[i]);
+    if(itc){
+      long lRemainingVol = itc->GetStorageVolume() - itc->GetContained()->GetVolume();
+      DBG3(lRemainingVol,itToStore->GetVolume(),itc->GetLabel().CStr());
+      if(lRemainingVol<itToStore->GetVolume())
+        continue;
+      
+      if(itc->IsAutoStoreMatch(itToStore->GetName(DEFINITE))){
```

Comment:
I think changing this line to
`if(!itc->IsLocked() && itc->IsAutoStoreMatch(itToStore->GetName(DEFINITE))){`
suffices to solve https://github.com/Attnam/ivan/pull/608#issuecomment-1000075184 request